.gitignore

```diff
@@ -5,3 +5,15 @@ RELEASE-VERSION
 dist
 *.egg-info
 __pycache__
+# thanks https://github.com/github/gitignore/blob/master/CMake.gitignore ! (under CC0 license) 
+CMakeLists.txt.user
+CMakeCache.txt
+CMakeFiles
+CMakeScripts
+Testing
+Makefile
+cmake_install.cmake
+install_manifest.txt
+compile_commands.json
+CTestTestfile.cmake
+_deps
```

Comment:
These should not be a problem with out-of-source builds, but ok

Reply:
agreed, I usually build out of build too. fixed it on autopilot..